test/suite/wttest.py

```diff
@@ -284,6 +312,42 @@ def skipTest(self, reason):
         self.skipped = True
         super(WiredTigerTestCase, self).skipTest(reason)
 
+    # Override unittest.run.
+    # We do this so we can intercept calls to the test method.
+    def run(self, result=None):
+        self._savedTestMethodName = self._testMethodName
+        self._testMethodName = "runTestMethod"
+        try:
+            super(WiredTigerTestCase, self).run(result)
+        finally:
+            self._testMethodName = self._savedTestMethodName
+
+    # This function is called in advance of every test method
+    # invocation.  It retries the test method whenever
+    # it fails due to a rollback error, up to a point.
+    def runTestMethod(self):
+        rollbacksAllowed = self._rollbacksAllowedPerTest
+        finished = False
+        method = getattr(self, self._savedTestMethodName)
+        WiredTigerTestCase._testsRun += 1
+        while not finished and rollbacksAllowed > 0:
```

Comment:
Should this be `rollbacksAllowed >= 0`?  Otherwise we'll never hit the `rollbacksAllowed == 0` condition in the `except` clause, below.

Reply:
Fixed.